Warnings during compilation

When compiling with `clang` the following warnings are shown up:
```
--- stderr: maliput                                            
/home/franco/Worskpace/RoadNetwork/tri_ws/src/maliput/maliput/src/utility/file_utils.cc:33:10: warning: moving a local object in a return statement prevents copy elision [-Wpessimizing-move]
  return std::move(filepaths);
         ^
/home/franco/Worskpace/RoadNetwork/tri_ws/src/maliput/maliput/src/utility/file_utils.cc:33:10: note: remove std::move call here
  return std::move(filepaths);
         ^~~~~~~~~~         ~
1 warning generated.
In file included from /home/franco/Worskpace/RoadNetwork/tri_ws/src/maliput/maliput/src/base/manual_discrete_value_rule_state_provider.cc:1:
/home/franco/Worskpace/RoadNetwork/tri_ws/src/maliput/maliput/include/maliput/base/manual_discrete_value_rule_state_provider.h:70:82: warning: 'DoGetState' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
  virtual std::optional<api::rules::DiscreteValueRuleStateProvider::StateResult> DoGetState(
                                                                                 ^
/home/franco/Worskpace/RoadNetwork/tri_ws/src/maliput/maliput/include/maliput/api/rules/discrete_value_rule_state_provider.h:35:38: note: overridden virtual function is here
  virtual std::optional<StateResult> DoGetState(const Rule::Id& id) const = 0;
                                     ^
/home/franco/Worskpace/RoadNetwork/tri_ws/src/maliput/maliput/src/plugin/maliput_plugin_manager.cc:21:10: warning: moving a local object in a return statement prevents copy elision [-Wpessimizing-move]
  return std::move(filepaths);
         ^
/home/franco/Worskpace/RoadNetwork/tri_ws/src/maliput/maliput/src/plugin/maliput_plugin_manager.cc:21:10: note: remove std::move call here
  return std::move(filepaths);
         ^~~~~~~~~~         ~
1 warning generated.
1 warning generated.
In file included from /home/franco/Worskpace/RoadNetwork/tri_ws/src/maliput/maliput/src/base/phase_based_right_of_way_discrete_value_rule_state_provider.cc:1:
In file included from /home/franco/Worskpace/RoadNetwork/tri_ws/src/maliput/maliput/include/maliput/base/phase_based_right_of_way_discrete_value_rule_state_provider.h:10:
/home/franco/Worskpace/RoadNetwork/tri_ws/src/maliput/maliput/include/maliput/base/manual_discrete_value_rule_state_provider.h:70:82: warning: 'DoGetState' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
  virtual std::optional<api::rules::DiscreteValueRuleStateProvider::StateResult> DoGetState(
                                                                                 ^
/home/franco/Worskpace/RoadNetwork/tri_ws/src/maliput/maliput/include/maliput/api/rules/discrete_value_rule_state_provider.h:35:38: note: overridden virtual function is here
  virtual std::optional<StateResult> DoGetState(const Rule::Id& id) const = 0;
                                     ^
1 warning generated.
In file included from /home/franco/Worskpace/RoadNetwork/tri_ws/src/maliput/maliput/test/base/phase_based_right_of_way_discrete_value_rule_state_provider_test.cc:1:
In file included from /home/franco/Worskpace/RoadNetwork/tri_ws/src/maliput/maliput/include/maliput/base/phase_based_right_of_way_discrete_value_rule_state_provider.h:10:
/home/franco/Worskpace/RoadNetwork/tri_ws/src/maliput/maliput/include/maliput/base/manual_discrete_value_rule_state_provider.h:70:82: warning: 'DoGetState' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
  virtual std::optional<api::rules::DiscreteValueRuleStateProvider::StateResult> DoGetState(
                                                                                 ^
/home/franco/Worskpace/RoadNetwork/tri_ws/src/maliput/maliput/include/maliput/api/rules/discrete_value_rule_state_provider.h:35:38: note: overridden virtual function is here
  virtual std::optional<StateResult> DoGetState(const Rule::Id& id) const = 0;
                                     ^
In file included from /home/franco/Worskpace/RoadNetwork/tri_ws/src/maliput/maliput/test/base/manual_discrete_value_rule_state_provider_test.cc:1:
/home/franco/Worskpace/RoadNetwork/tri_ws/src/maliput/maliput/include/maliput/base/manual_discrete_value_rule_state_provider.h:70:82: warning: 'DoGetState' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
  virtual std::optional<api::rules::DiscreteValueRuleStateProvider::StateResult> DoGetState(
                                                                                 ^
/home/franco/Worskpace/RoadNetwork/tri_ws/src/maliput/maliput/include/maliput/api/rules/discrete_value_rule_state_provider.h:35:38: note: overridden virtual function is here
  virtual std::optional<StateResult> DoGetState(const Rule::Id& id) const = 0;
```


The following warnings are related to https://github.com/ToyotaResearchInstitute/maliput/issues/384
```
1 warning generated.
/home/franco/Worskpace/RoadNetwork/tri_ws/src/maliput/maliput/test/plugin/tools/lorem_ipsum_test_plugin.cc:16:24: warning: 'GetMaliputPluginId' has C-linkage specified, but returns user-defined type 'std::string' (aka 'basic_string<char>') which is incompatible with C [-Wreturn-type-c-linkage]
extern "C" std::string GetMaliputPluginId() {
                       ^
/home/franco/Worskpace/RoadNetwork/tri_ws/src/maliput/maliput/test/plugin/tools/lorem_ipsum_test_plugin.cc:28:24: warning: 'GetAString' has C-linkage specified, but returns user-defined type 'std::string' (aka 'basic_string<char>') which is incompatible with C [-Wreturn-type-c-linkage]
extern "C" std::string GetAString() { return "LoremIpsum"; };
                       ^
2 warnings generated.
/home/franco/Worskpace/RoadNetwork/tri_ws/src/maliput/maliput/test/plugin/tools/multiply_integers_test_plugin.cc:16:24: warning: 'GetMaliputPluginId' has C-linkage specified, but returns user-defined type 'std::string' (aka 'basic_string<char>') which is incompatible with C [-Wreturn-type-c-linkage]
extern "C" std::string GetMaliputPluginId() { return std::string("multiply_integers_test_plugin"); }
                       ^
1 warning generated.
/home/franco/Worskpace/RoadNetwork/tri_ws/src/maliput/maliput/test/plugin/tools/sum_integers_test_plugin.cc:16:24: warning: 'GetMaliputPluginId' has C-linkage specified, but returns user-defined type 'std::string' (aka 'basic_string<char>') which is incompatible with C [-Wreturn-type-c-linkage]
extern "C" std::string GetMaliputPluginId() { return std::string("sum_integers_test_plugin"); }
                       ^
```

--- a/maliput/include/maliput/base/manual_discrete_value_rule_state_provider.h
+++ b/maliput/include/maliput/base/manual_discrete_value_rule_state_provider.h
@@ -68,7 +68,7 @@ class ManualDiscreteValueRuleStateProvider : public api::rules::DiscreteValueRul
   // benefit from injecting their own getter and then forwarding calls to this
   // function.
   virtual std::optional<api::rules::DiscreteValueRuleStateProvider::StateResult> DoGetState(
-      const api::rules::Rule::Id& id) const;
+      const api::rules::Rule::Id& id) const override;
 
  private:
   // @throws common::assertion_error When @p state is unrecognized in

--- a/maliput/src/plugin/maliput_plugin_manager.cc
+++ b/maliput/src/plugin/maliput_plugin_manager.cc
@@ -18,7 +18,7 @@ std::vector<std::string> GetPluginLibraryPaths(const std::string& env_var) {
     const auto filepaths_from_dir = maliput::utility::GetAllFilePathsFromDirectory(path, "so");
     filepaths.insert(filepaths.end(), filepaths_from_dir.begin(), filepaths_from_dir.end());
   }
-  return std::move(filepaths);
+  return filepaths;
 }
 
 }  // namespace

--- a/maliput/src/utility/file_utils.cc
+++ b/maliput/src/utility/file_utils.cc
@@ -30,7 +30,7 @@ std::vector<std::string> GetAllFilePathsFromDirectory(const std::string& directo
     }
     filepaths.push_back(entry.path());
   }
-  return std::move(filepaths);
+  return filepaths;
 }
 
 std::vector<std::string> GetAllPathsFromEnvironment(const std::string& env_var) {

